src/compilers/variable.cr

```diff
@@ -96,12 +96,19 @@ module Mint
           else
             "SHOULD NEVER HAPPEN"
           end
-        when Tuple(Ast::Node, Int32)
+        when Tuple(Ast::Node, Int32), Tuple(Ast::Node, Array(Int32) | Int32)
           case item = entity[0]
           when Ast::WhereStatement, Ast::Statement
             case target = item.target
             when Ast::TupleDestructuring
-              js.variable_of(target.parameters[entity[1]])
+              case val = entity[1]
+              when Int32
+                js.variable_of(target.parameters[val])
+              when Array(Int32)
+                js.variable_of(val.reduce(target) do |curr_type, curr_val|
+                  curr_type.as(Ast::TupleDestructuring).parameters[curr_val]
+                end)
+              end.not_nil!
```

Comment:
@Sija which is more idiomatic? 

```
end.not_nil!
``` 

or

```
else
  raise "Cannot Happen!"
end
```

Reply:
In cases like these - in which I'm pretty sure the value is there - I'd use `not_nil!` since the stacktrace should be enough (except for cases in which we build mint with `--no-debug` flag - but then we're covered as long as the bug is reproducible).

Reply:
Got it thanks :+1: